### DISCUSSION TITLE
Upcoming changes to 'Update' button functionality

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1474] [Plugin Development]
TL;DR - Are any plugins relying on telling their users "press the main Update button" in order to trigger some of their own functionality ?  *If they are, they need to speak up* **NOW**.

Project for this: https://github.com/EDCD/EDMarketConnector/projects/17

1. Users keep on thinking that pressing the 'Update' button does things that it does not.  The only things this achieves with core-only EDMC are:
    1. Trigger a pull of CAPI `/profile`, and if docked `/market` and `/shipyard` as well.
    2. That data is then utilised in the 'File' > 'Status' popup.
    3. The core plugins have their `cmdr_data()` called.   For EDDN this will send commodity, outfitting and shipyard messages as appropriate.   As of EDMC 5.3.0 the most that any other core plugin does with this is keep the System and Station links in the main UI up to date.  The core Inara plugin no longer updates 'Commander Wallet balance' from this (as there's no 'Total Wealth' figure available in this data, and tracking differences otherwise is fraught with missing data).
    4. The resulting data is sent to plugins in the `cmdr_data()` call-in.

    But users persist in thinking that it will, e.g. send the entire current state of their Commander to Inara.  It does not.  Nor is it going to have *any* effect on Journal-sourced data (e.g. update flight log on EDSM).
    There are also edge and corner cases around pressing 'Update' when, e.g. the game isn't even running, or you last weren't docked.  Investigating any such just wastes time when it is unintended use.

2. So I would like to change the label and functionality of the 'Update' button to better reflect its intended purpose.  **The only expectation should be that pressing the 'Update' button whilst docked somewhere will attempt to send information about that location to EDDN.**

3. In order to ensure 'File' > 'Status' always works we will also add an 'Auto update on login' option which will work even if not docked.  Thus all plugins can assume they'll get `cmdr_data()` called shortly after game login (if the user turns this off, then that's their problem).

4. The Settings > Output > "Automatically update on docking" option will continue to function as it currently does.  This is precisely the intended functionality.

5. After login to the game if you're not docked then the 'Update' button will be disabled, as will the hotkey.  The game UI will re-label the button to indicate it is disabled (and if I can figure out how, cause mouseover to show the 'stop' symbol').

I'm already checking through all the plugins listed on [the EDMC Wiki:Plugins page](https://github.com/EDCD/EDMarketConnector/wiki/Plugins) and contacting authors if needs be.

If your plugin isn't yet listed on that page then please consider the information in this post and reply as needed.

Remember, for many situations in which you might be relying on CAPI data passed into `cmdr_data()` there's probably a Journal event you could better react to in `journal_entry()`, e.g. there's no point in tracking current system via `cmdr_data()` when you can just do it from Journal events such as: Startup (the actual journal event), StartUp (this is the EDMC synthesized event generated when EDMC detects the game was already running when *it* started), Location, FSDJump, CarrierJump.

**Comment by PsymonsB** [comment]
Our plugin (huttonhelper) doesn't rely on the update button or capi, and its often user misperception of what this does as it appears below our plugin (in some instances), could it be hidden in the menus and not the main gui as its use is quite specific
